File: 1863-sum-of-all-subset-xor-totals/1863-sum-of-all-subset-xor-totals.cpp
class Solution {
public:
    int subsetXORSum(vector<int>& nums) {
        int n = nums.size();

        int res = 0;
        for (int mask = 0; mask < (1 << n); mask++) {
            int tempXr = 0;
            for (int bit = 0; bit < n; bit++) {
                if ((mask >> bit) & 1) {
                    tempXr = tempXr ^ nums[bit];
                }
            }
            res += tempXr;
        }
        
        return res;
    }
};

File: 1863-sum-of-all-subset-xor-totals/1863-sum-of-all-subset-xor-totals.java
class Solution {
    public int subsetXORSum(int[] nums) {
        int len = nums.length;

        int ans = 0;
        for (int i = 0; i < (1 << len); i++) {
            int currXor = 0;
            for (int bit = 0; bit < len; bit++) {
                if ((i & (1 << bit)) > 0) {
                    currXor = currXor ^ nums[bit];
                }
            }

            ans += currXor;
        }

        return ans;
    }
}